src/app/data-model/DecodeWithoutFabricIndex.h

```diff
@@ -0,0 +1,50 @@
+/*
+ *
+ *    Copyright (c) 2020-2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/data-model/DecodableList.h>
+#include <app/data-model/Decode.h>
+
+namespace chip {
+namespace app {
+namespace DataModel {
+
+template <typename X,
+          typename std::enable_if_t<std::is_class<X>::value &&
+                                        std::is_same<decltype(Decode(std::declval<TLV::TLVReader &>(), std::declval<X &>(),
+                                                                     std::declval<Optional<FabricIndex> &>())),
+                                                     CHIP_ERROR>::value,
+                                    X> * = nullptr>
+CHIP_ERROR DecodeWithoutFabricIndex(TLV::TLVReader & reader, X & x)
```

Comment:
When would this be used?  What does it mean to use it?  Needs _very_ careful documentation.